### DISCUSSION TITLE
DMA on SPI bus for Polarfire SoC

### DISCUSSION THREAD

**Discussion by fcuzzocrea** [polarfire-soc/polarfire-soc-documentation#357] [Bare metal embedded software]
Is it possible to use DMA on the physical SPI bus on the Polarfire SoC ?

**Comment by eppidei** [comment]
Hi @fcuzzocrea , that's not possible. the MSS SPI provides a FIFO interface but the Platform DMA engine supports only memory mapped transactions. The way to implement a SPI DMA is to go through the soft SPI in the fabric and access memory using the FIC routes.

**Reply by PrajinIr** [reply]
Is there any example code available for this in the repo.

**Reply by PrajinIr** [reply]
How can we configure this SPI DMA
